src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!this.isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
So, this works only with the `AllItemsIterator`, but not with `BusyItemsIterator`, right?